### PR TITLE
[Trimming] Enable trimming and AOT analyzers

### PR DESCRIPTION
**Pull request by simonrozsival**
### Description of Change

This PR enables trimming, AOT, and single file analyzers to assemblies where it doesn't require any changes. For Essentials.csproj and Compatibility.csproj I changed IsTrimmable to IsAotCompatible.

The only projects that don't have the analyzers enabled yet are:
- Controls.Core.csproj - will need addressing a few issues
- Controls.Xaml.csproj - will need addressing a few issues

I'm also skipping adding analyzers to:
- Test projects
- Compatibility projects
- ControlGallery projects
- *.Design projects (net472)
- Templates
- netstandard2.0 projects

### Issues Fixed

Contributes to #18658

### CODE REVIEW

**Review by jonathanpeppers — APPROVED** [review]

File: src/BlazorWebView/src/WindowsForms/BlazorWebView.cs [hunk 1]
```diff
@@ -170,7 +170,9 @@ private void StartWebViewCoreIfPossible()
 			// We assume the host page is always in the root of the content directory, because it's
 			// unclear there's any other use case. We can add more options later if so.
 			string appRootDir;
+#pragma warning disable IL3000 // 'System.Reflection.Assembly.Location.get' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.
 			var entryAssemblyLocation = Assembly.GetEntryAssembly()?.Location;
+#pragma warning restore IL3000
```
Comment (jonathanpeppers) [inline src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:175]: @simonrozsival maybe we could file an issue about this one?

Comment (simonrozsival) [reply src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:175]: Just to make sure: do you mean changing `Environment.CurrentDirectory` to `System.AppContext.BaseDirectory` or suppressing the warning for getting the assembly location?

Comment (jonathanpeppers) [reply src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:175]: An issue to change it to `System.AppContext.BaseDirectory`, as it seems more appropriate. You can publish WinForms and WPF as single file, and `Environment.CurrentDirectory` doesn't seem correct.

Comment (simonrozsival) [reply src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:175]: https://github.com/dotnet/maui/issues/21750

**Review by StephaneDelcroix — APPROVED** [review]

**Comment by jonathanpeppers** [comment]
@Eilon or someone in @dotnet/dotnet-maui-blazor-eng care to review?

This ignores a trimmer warning in WinForms/WPF BlazorWebView, but we filed an issue for the future at:
* https://github.com/dotnet/maui/issues/21750

**Comment by Eilon** [comment]
> @Eilon or someone in @dotnet/dotnet-maui-blazor-eng care to review?
> 
> This ignores a trimmer warning in WinForms/WPF BlazorWebView, but we filed an issue for the future at:
> 
> * [BlazorWebView: Use AppContext.BaseDirectory instead of Environment.CurrentDirectory  #21750](https://github.com/dotnet/maui/issues/21750)

I'm reviewing. If we take the change in https://github.com/dotnet/maui/pull/21797, does this PR need to change at all? (I think the change in https://github.com/dotnet/maui/pull/21797 is good, but I want to test locally first)

**Comment by jonathanpeppers** [comment]
This PR doesn't need to change if https://github.com/dotnet/maui/pull/21797 goes in.

The warning suppressed here is around `Assembly.Location` that returns `null` in self-contained apps. But the code already does a null check.

**Pull request merged**